src/System.IO.Pipelines/System/IO/Pipelines/WritableBuffer.cs

```diff
@@ -76,9 +67,52 @@ public void Commit()
         /// Will <see cref="Commit"/> if necessary.
         /// </summary>
         /// <returns>A task that completes when the data is fully flushed.</returns>
-        public WritableBufferAwaitable FlushAsync(CancellationToken cancellationToken = default)
+        public Awaitable<FlushResult> FlushAsync(CancellationToken cancellationToken = default)
         {
             return Pipe.FlushAsync(cancellationToken);
         }
+
+        /// <summary>
+        /// Writes the source <see cref="ReadOnlySpan{Byte}"/> to the <see cref="WritableBuffer"/>.
+        /// </summary>
+        /// <param name="buffer">The <see cref="WritableBuffer"/></param>
+        /// <param name="source">The <see cref="ReadOnlySpan{Byte}"/> to write</param>
+        public void Write(ReadOnlySpan<byte> source)
```

Comment:
Why isn’t this an extension method?

Reply:
@KrzysztofCwalina want's fewer classes and extension methods.